include/mxnet/lib_api.h

```diff
@@ -982,17 +1045,22 @@ extern "C" {
     // create a vector of tensors for inputs
     std::vector<MXTensor> inputs(num_in);
     for (int i = 0; i < num_in; i++) {
-      inputs[i].setTensor(indata[i], (MXDType)intypes[i], inshapes[i], indims[i], inIDs[i]);
+      std::string ctx_str(indev_type[i]);
+      MXContext inctx = {ctx_str, indev_id[i]};
+      inputs[i].setTensor(indata[i], (MXDType)intypes[i], inshapes[i], indims[i],
```

Comment:
string can be initialized with char* http://www.cplusplus.com/reference/string/string/string/ (see constructor 4)

can we just inline everything and not create intermediary variables?
```
inputs[i].setTensor(indata[i], (MXDType)intypes[i], inshapes[i], indims[i], inIDs[i], {ctx_str, indev_id[i]});
```